include/fluent-bit/flb_downstream.h

```diff
@@ -40,6 +41,7 @@ struct flb_downstream {
     flb_sockfd_t           server_fd;
     struct flb_connection *dgram_connection;
 
+    pthread_mutex_t        busy_queue_mutex;
```

Comment:
This is for following data race.

```
==================
WARNING: ThreadSanitizer: data race (pid=47741)
  Write of size 8 at 0x7b3800010048 by thread T3:
    #0 __mk_list_add /home/taka/git/fluent-bit/lib/monkey/include/monkey/mk_core/mk_list.h:60 (fluent-bit+0x144c66)
    #1 mk_list_add /home/taka/git/fluent-bit/lib/monkey/include/monkey/mk_core/mk_list.h:65 (fluent-bit+0x144cbc)
    #2 flb_downstream_conn_get /home/taka/git/fluent-bit/src/flb_downstream.c:299 (fluent-bit+0x14a6f4)
    #3 in_fw_collect /home/taka/git/fluent-bit/plugins/in_forward/fw.c:130 (fluent-bit+0x49a274)
    #4 input_collector_fd /home/taka/git/fluent-bit/src/flb_input_thread.c:168 (fluent-bit+0xe1395)
    #5 engine_handle_event /home/taka/git/fluent-bit/src/flb_input_thread.c:183 (fluent-bit+0xe2111)
    #6 input_thread /home/taka/git/fluent-bit/src/flb_input_thread.c:384 (fluent-bit+0xe2111)
    #7 step_callback /home/taka/git/fluent-bit/src/flb_worker.c:43 (fluent-bit+0x158b28)

  Previous read of size 8 at 0x7b3800010048 by thread T1:
    #0 flb_downstream_conn_timeouts /home/taka/git/fluent-bit/src/flb_downstream.c:418 (fluent-bit+0x14abd6)
    #1 cb_engine_sched_timer /home/taka/git/fluent-bit/src/flb_engine.c:172 (fluent-bit+0x124a94)
    #2 flb_sched_event_handler /home/taka/git/fluent-bit/src/flb_scheduler.c:428 (fluent-bit+0x13ad70)
    #3 flb_engine_start /home/taka/git/fluent-bit/src/flb_engine.c:945 (fluent-bit+0x127254)
    #4 flb_lib_worker /home/taka/git/fluent-bit/src/flb_lib.c:629 (fluent-bit+0xa1ced)
```

Reply:
It points touching `&stream->busy_queue`.
I added a mutex for touching it.

https://github.com/fluent/fluent-bit/blob/v2.0.9/src/flb_downstream.c#L299
and
https://github.com/fluent/fluent-bit/blob/v2.0.9/src/flb_downstream.c#L418